packages/hub/routes/card-spaces.ts

```diff
@@ -54,36 +59,74 @@ export default class CardSpacesRoute {
       return;
     }
 
+    if (!validateRequiredFields(ctx, { requiredRelationships: ['merchant-info'] })) {
+      return;
+    }
+
     const cardSpace: CardSpace = {
       id: shortUuid.uuid(),
-      url: this.sanitizeText(ctx.request.body.data.attributes['url']),
+      merchantId: this.sanitizeText(ctx.request.body.data.relationships['merchant-info'].data.id),
       profileName: this.sanitizeText(ctx.request.body.data.attributes['profile-name']),
       profileDescription: this.sanitizeText(ctx.request.body.data.attributes['profile-description']),
       profileCategory: this.sanitizeText(ctx.request.body.data.attributes['profile-category']),
       profileButtonText: this.sanitizeText(ctx.request.body.data.attributes['profile-button-text']),
       profileImageUrl: this.sanitizeText(ctx.request.body.data.attributes['profile-image-url']),
       profileCoverImageUrl: this.sanitizeText(ctx.request.body.data.attributes['profile-cover-image-url']),
-      ownerAddress: ctx.state.userAddress,
     };
 
-    let errors = await this.cardSpaceValidator.validate(cardSpace);
-    let hasErrors = Object.values(errors).flatMap((i) => i).length > 0;
+    let merchantInfoError;
+
+    if (cardSpace.merchantId) {
```

Comment:
Shouldn't this attribute always be present? Maybe we can pass the `cardSpace` into the validator at this place and let it figure out whether merchantId is present or not and return an error if it's not?

Reply:
`validateRequiredFields` will indeed actually catch this, removed now, thanks.